esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -2916,28 +2916,36 @@ bool CFileSprayEx::onFileList(IEspContext &context, IEspFileListRequest &req, IE
             StringBuffer dzName;
             if (isEmptyString(dropZoneName))
                 dropZoneName = findDropZonePlaneName(netaddr, sPath, dzName);
-            if (!isEmptyString(dropZoneName))
+            Owned<IPropertyTree> dropZone;
+            if (!isEmptyString(dropZoneName)) // must be true, unless bare-metal and isDropZoneRestrictionEnabled()==false
+            {
+                dropZone.setown(getDropZonePlane(dropZoneName));
+                if (!dropZone)
+                    throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "Unknown landing zone: %s", dropZoneName);
+                if (!isPathInPlane(dropZone, sPath))
+                    throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "Path '%s' is not valid for dropzone '%s'", sPath.str(), dropZoneName);
```

Comment:
Also, now you've consolidated the checks into checkDropZoneInputAndAccess, it looks like this code is very similar and checkDropZoneInputAndAccess should be used here too.

Reply:
> Where is "localhost" coming from in this context?

The "localhost" comes from:
static void gatherDropZoneMachines(IArrayOf<IEspTpMachine> & tpMachines, IPropertyTree & plane)
{
    const char * prefix = plane.queryProp("@prefix");
    if (plane.hasProp("hosts"))
        gatherDropZoneMachinesFromHosts(tpMachines, plane, prefix);
    else
        tpMachines.append(*createHostTpMachine("localhost", prefix));
}

> I don't think that doesn't sound right.

The issue is because the getPlaneHostGroup() returns nullptr for my local Docker Desktop:
<planes category="lz" name="mydropzone" prefix="/var/lib/HPCCSystems/mydropzone"/>
What should I do to avoid this issue?